dev-notes/minimal-os-for-fobnail.md

```diff
@@ -0,0 +1,718 @@
+bootloader called
+[Gigaboot](https://fuchsia.googlesource.com/fuchsia/+/refs/heads/main/src/firmware/gigaboot/).
+Multiboot 1 support is still present, but SKL doesn't support it anyway. See
+[Zephyr](#running-in-dlme) section above for details.
+
+#### Fobnail integration
+
+Fuchsia provides a Unix-like environment with musl C library. Network is
+available but there is no EEM driver, and only XHCI host driver is supported.
+Since EHCI is still widely used, this is a major limitation. TPM driver is
+available but without `libtss2`.
+
+Zircon has a kexec-like capability which allows to boot another Zircon instance.
+This is known as `mexec`.
+
+Following things would have to be done:
+
+* Bring `libtss2` stack to Zircon.
+* Bring EEM driver. There is ECM driver available, which could serve as
+  reference.
+* Bring at least USB EHCI driver, ideally should support UHCI and OHCI too.
+* Figure out how to use `mexec` to boot Linux.
+
+### Xen
+
+Xen is a bare-metal hypervisor available on x86 and ARM. It is able to run Linux
+in a paravirtualized environment. Xen contains only drivers absolutely necessary
+for hypervisor and VM to boot. Control over most of the hardware is handed to
+the first VM known as [Dom0](https://wiki.xenproject.org/wiki/Dom0). Usuallly
+Linux runs in Dom0, there are other OSes which also support running in Dom0
+(BSD, Solaris), however there is no advantage of using them over Linux. Xen also
+has ability to run in dom0-less mode - bootloader provides VM images and Xen
+starts VMs on their own (see section below for details).
+
+#### Running in DLME
+
+Xen is supported by Trenchboot's GRUB and can run without any modifications.
+See [xen-in-dlme.md](xen-in-dlme.md) for demo.
+
+#### Integration with Fobnail
+
+Domain 0 less mode eliminates need for trusted VM. Also, since Xen itself is
+responsible for booting multiple VMs there is no need for tools that manage Xen
+(which don't work on any OS). Fobnail Attester could be integrated in a similar
+way that with CAmkES based VM (see CAmkES section above for details). The main
+difference would be that the secure component would be a separate VM.
+
+Unfortunatelly, Xen is bloated and it doesn't meet the "minimal" requirement.
+This is a major problem.
+
+## Zephyr PoC test
+
+In addition to conducting the theoretical analysis, it was decided to include a
+test report of one of the operating systems in order to verify the information
+presented. The choice fell on Zephyr.
+
+Running DLME requires GRUB from TrenchBoot as mainline doesn't have DLME
+support. Currently, there is an ongoing discussion how TrenchBoot should be
+integrated into Linux - see the following links:
+
+* https://groups.google.com/g/trenchboot-devel/c/-RxTtan5H3I/m/vU-yp5kHAgAJ,
+* https://groups.google.com/g/trenchboot-devel/c/QLMg0r4XTcs/m/YrN2bX3PDQAJ,
+* https://groups.google.com/g/trenchboot-devel/c/6ilwQUkwt9w/m/dpyriBCHAgAJ.
```

Comment:
I don't like the way it is linked, please provide title of patch series as link name and hide url which say nothing about the content behind.

Reply:
Fixed in seperate doc [cdb888f](https://github.com/fobnail/docs/pull/26/commits/cdb888f92ba674a5c52d1d226a10c03b75cce94b#diff-277d2cf31e610e3342b956bc2893797b89598425b3fef709288e9cadd09c60d1R11)